frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,511 @@
+    Service: "Service",
+    Date: "Date",
+    ReportId: "Report Id",
+    Main: "Main",
+    Sub: "Sub",
+    FileName: "File Name",
+    TransportResults: "Transport Results",
+};
+
+const FilterOptionsEnum = {
+    NONE: "none",
+    ASC: "asc",
+    DESC: "desc",
+};
+
+const StatusEnum = {
+    BATCH: "BATCH",
+    PROCESS: "PROCESS",
+    READY: "READY",
+};
+
+const MessageReceiversCol = ({
+    columnHeaderTitle,
+    activeColumn,
+    setActiveColumn,
+    activeColumnSortOrder,
+    setActiveColumnSortOrder,
+    rowSpan,
+}: MessageReceiversColProps) => {
+    const isCurrentlyActiveColumn = columnHeaderTitle === activeColumn;
+    const handleColHeaderClick = () => {
+        if (!isCurrentlyActiveColumn) {
+            // Reset active column and sort order on new column click
+            setActiveColumn(columnHeaderTitle);
+            setActiveColumnSortOrder(FilterOptionsEnum.NONE);
+        }
+
+        // Explicitly set the proceeding sort order
+        if (activeColumnSortOrder === FilterOptionsEnum.NONE) {
+            setActiveColumnSortOrder(FilterOptionsEnum.ASC);
+        } else if (activeColumnSortOrder === FilterOptionsEnum.ASC) {
+            setActiveColumnSortOrder(FilterOptionsEnum.DESC);
+        } else if (activeColumnSortOrder === FilterOptionsEnum.DESC) {
+            setActiveColumnSortOrder(FilterOptionsEnum.NONE);
+        }
+    };
+    const sortIcon = useMemo(() => {
+        if (
+            isCurrentlyActiveColumn &&
+            activeColumnSortOrder === FilterOptionsEnum.ASC
+        ) {
+            return <Icon.ArrowUpward />;
+        } else if (
+            isCurrentlyActiveColumn &&
+            activeColumnSortOrder === FilterOptionsEnum.DESC
+        ) {
+            return <Icon.ArrowDownward />;
+        }
+        return <Icon.SortArrow />;
+    }, [activeColumnSortOrder, isCurrentlyActiveColumn]);
+    return (
+        <th
+            role="button"
+            className={classnames({
+                "active-col-header":
+                    isCurrentlyActiveColumn &&
+                    activeColumnSortOrder !== FilterOptionsEnum.NONE,
+            })}
+            onClick={handleColHeaderClick}
+            rowSpan={rowSpan}
+        >
+            <div className="column-header-title-container">
+                <p
+                    className={classnames({
+                        "text-pre":
+                            columnHeaderTitle === ColumnDataEnum.FileName,
+                    })}
+                >
+                    {columnHeaderTitle}
+                </p>
+                {sortIcon}
+            </div>
+        </th>
+    );
+};
+
+const MessageReceiversRow = ({
+    receiver,
+    activeColumn,
+    activeColumnSortOrder,
+    setModalText,
+    modalRef,
+}: MessageReceiversRowProps) => {
+    // We highlight the entire active column so within the row,
+    // we need to know which column is active
+    const checkForActiveColumn = (colName: string) =>
```

Comment:
Small point, but the name of this function is a little misleading.  Could it be named something like `getColumnClasses`?

Reply:
Yeah, sounds (semantically) good!